dlk/python/tools/test_debug.py

```diff
@@ -53,19 +54,34 @@ def main(debug_data_path, expected_data_path):
     if not expected_output:
         print(f"Expected data path {e_output} is empty (no .npy files)")
 
+    results = {}
+    diffs = {}
     for o in debug_output:
         name_dbg = ptrn_dbg.match(o.stem).group(1)
+        output_id_dbg = ptrn_dbg.match(o.stem).group(2)
         for eo in expected_output:
-            name_ex = ptrn_ex.match(eo.stem).group(1)
-            output_id_ex = ptrn_ex.match(eo.stem).group(2)
-            if name_ex == name_dbg and output_id_ex == '0':
+            name_ex = ptrn_ex.match(eo.stem).group(2)
+            output_id_ex = ptrn_ex.match(eo.stem).group(3)
+            if name_ex == name_dbg and output_id_ex == output_id_dbg:
                 data_dbg = np.load(o)
                 data_ex = np.load(eo)
 
-                if np.allclose(data_ex.flatten() * data_dbg['scale'], data_dbg['data'], rtol=0.00001, atol=0.00001):
-                    print(f"[OK]   {name_ex}")
+                r_tol = 0.0001
+                a_tol = 0.0001
+
+                idx = np.isclose(data_ex.flatten() * data_dbg['scale'], data_dbg['data'], rtol=r_tol, atol=a_tol)
+
+                if np.all(idx):
+                    results[ptrn_ex.match(eo.stem).group(1)] = f"[OK]   {name_ex}:{output_id_ex}"
                 else:
-                    print(f"[FAIL] {name_ex}")
+                    results[ptrn_ex.match(eo.stem).group(1)] = f"[FAIL] {name_ex}:{output_id_ex}"
+
+                diffs[ptrn_ex.match(eo.stem).group(1)] = data_ex.flatten().size - np.count_nonzero(idx)
+
+    sorted_results = [val for key, val in sorted(results.items())]
+    sorted_diffs = [val for key, val in sorted(diffs.items())]
+    for r, d in zip(sorted_results, sorted_diffs):
+        print(r)
```

Comment:
don't print within unittests. Ret the tetst fail

Reply:
This file is not for unittest. Maybe it's better to rename the filename?

Reply:
This file is not for unittest. Maybe it's better to rename the filename?

Reply:
The file name is indeed too confusing, I will rename it.